pkg/schemas/task_gen.go

```diff
@@ -102,19 +129,39 @@ func TaskToSchema(task *sdk.Task) map[string]any {
 	taskSchema["schema_name"] = task.SchemaName
 	taskSchema["owner"] = task.Owner
 	taskSchema["comment"] = task.Comment
-	taskSchema["warehouse"] = task.Warehouse
+	if task.Warehouse != nil {
+		taskSchema["warehouse"] = task.Warehouse.Name()
+	}
 	taskSchema["schedule"] = task.Schedule
-	taskSchema["predecessors"] = task.Predecessors
+	taskSchema["predecessors"] = collections.Map(task.Predecessors, sdk.SchemaObjectIdentifier.FullyQualifiedName)
 	taskSchema["state"] = string(task.State)
 	taskSchema["definition"] = task.Definition
 	taskSchema["condition"] = task.Condition
 	taskSchema["allow_overlapping_execution"] = task.AllowOverlappingExecution
-	taskSchema["error_integration"] = task.ErrorIntegration
+	if task.ErrorIntegration != nil {
+		taskSchema["error_integration"] = task.ErrorIntegration.Name()
+	}
 	taskSchema["last_committed_on"] = task.LastCommittedOn
 	taskSchema["last_suspended_on"] = task.LastSuspendedOn
 	taskSchema["owner_role_type"] = task.OwnerRoleType
 	taskSchema["config"] = task.Config
 	taskSchema["budget"] = task.Budget
+	taskSchema["last_suspended_reason"] = task.LastSuspendedReason
+	finalizer := ""
+	if task.TaskRelations.FinalizerTask != nil {
+		finalizer = task.TaskRelations.FinalizerTask.FullyQualifiedName()
```

Comment:
Hmm why is this handled in a different way than warehouse?

Reply:
It had to be adjusted manually, added comment and an entry in readme for schema generator